docs/API References.md

```diff
@@ -7,4 +7,187 @@ We'll glad to see all your questions:
+    * [Client](#-client)
+    * [Common](#-common)
+    * [Configuration](#-configuration)
+    * [Functional tests](#-functional-tests)
+* [Configuration Files](#-configuration-files)
+    * [Client Config](#client-config)
+    * [Server Config](#server-config)
+    * [Build Config](#build-config)
+* [Authorization](#authorization)
+* [Environment Variables](#-environment-variables)
+    * [URL Settings](#url-settings)
+    * [Custom Environment Variables](#custom-environment-variables)
+
+## **Available Scripts**
+
+In the project directory, you can run:
+
+### `npm run dev`
+Runs the app in the development mode.
+
+Two web servers are  started: one - for the frontend/UI part, based on the webpack-dev-server on the 3001 port by default, and another one - for the API backend part to provide API for reSolve endpoints, based on express on the 3000 port. Development servers provide all the required debugging capabilities, including [Hot Module Replacement](https://webpack.js.org/concepts/hot-module-replacement/) and [source maps](https://webpack.js.org/configuration/devtool/).
+
+Open [http://localhost:3000](http://localhost:3000/) or http://<your_ip>:3000 to view the app in the browser (you can change your [url settings](#url-settings)).
+
+### `npm run build`
+Builds client and server bundles for production through Webpack.
+
+Building is performed in the `NODE_ENV === 'production'` [mode](https://webpack.js.org/guides/production/#node-environment-variable), so the build is optimized. No additional HTTP server for the serving client bundle and assets are  built.
+
+### `npm start`
+Runs the built app in the production mode.
+
+Open [http://localhost:3000](http://localhost:3000/) or http://<your_ip>:3000 to view it in the browser (you can change your [url settings](#url-settings).
+
+### `npm run update [version]`
+Updates all resolve packages to the latest version according to [semver](https://docs.npmjs.com/getting-started/semantic-versioning).
+If the `version` argument is set, the command updates packages to the specified version.
+
+## **Project Structure Overview**
+[Create ReSolve App](https://www.npmjs.com/package/creat-resolve-app) is an NPM package referencing the latest [reSolve framework package](../..#) versions. It consists of the common isomorphic part which describes domain business logic and React components for the presentation. No implicit server part is needed - it is encapsulated in `resolve-scripts`, but can be customized using [config](#-configuration-files). The project also includes unit & E2E testing and deployment assets. All source code and functional tests are written in the [ES2016](http://2ality.com/2016/01/ecmascript-2016.html).
+
+```
+📁 resolve-app
+    📄 .gitignore
+    📄 .yarnrc
+    📄 LICENSE
+    📄 README.md
+    📄 package-lock.json
+    📄 package.json
+    📄 resolve.build.config.js
+    📄 resolve.client.config.js
+    📄 resolve.server.config.js
+    📁 client/
```

Comment:
add containers, sagas and other optional foders from react and redux projects

Reply:
Okay, we have developed a new default project structure

client/
   components/   
   containers/
      App.js
   reducers/
   store/
   routes.js
common/
      aggregates/
      read-models/
      view-models/
tests/
config.js
...
other files